analyses/molecular-subtyping-MB/02-compare-classes.Rmd

```diff
@@ -0,0 +1,221 @@
+---
+title: "Comparison of Expected and Observed MB Subtype Classification"
+output:
+  html_document:
+    df_print: paged
+params:
+  expected_input:
+    value: input/expected_class.rds
+  observed_input:
+    value: results/mb-classified.rds
+---
+  
+```{r include = FALSE}
+knitr::opts_chunk$set(comment = NA)
+getOption(x = 'DT.warn.size', default = FALSE)
+```
+
+```{r load_packages, echo = TRUE}
+# load libraries
+suppressPackageStartupMessages(library(tidyverse))
+suppressPackageStartupMessages(library(knitr))
+suppressPackageStartupMessages(library(DT))
+```
+
+```{r read_data, echo = TRUE}
+# read input
+exp.class <- readRDS('input/expected_class.rds')
+obs.class <- readRDS('results/mb-classified.rds')
+
+# format ambiguous expected classes 
+exp.class <- exp.class %>%
+  mutate(pathology_subtype = replace(pathology_subtype, 
+                                     pathology_subtype == "Group 3 or 4", "Group3, Group4")) %>%
+  mutate(pathology_subtype = gsub(" ", "", pathology_subtype))
```

Comment:
So for this, there is `Group 3` and `Group 4` in path reports and we have `Group3` and `Group4`  which is why the `gsub`.

Reply:
Ah, so `pathology_subtype` can have the values `Group 3 or 4` or `Group 3, Group 4` before you do any mutating - is that correct?

Reply:
Yes these are the unique values for expected types (pathology):

> unique(dat$pathology_subtype)
[1] NA             "WNT"          "Group 3 or 4" "SHH"          "non-WNT"     
[6] "Group 4" 

So I convert `Group 3 or 4` to `Group3, Group4` (so that we can match either Group3 or Group4 predicted subtype) and gsub converts `Group 4` to `Group4` (because for observed types we have SHH, WNT, Group3 and Group4 i.e. no spaces)